lib/provider.js

```diff
@@ -248,6 +249,7 @@ class Provider extends events.EventEmitter {
     Object.assign(session, {
       account,
       loginTs: ts,
+      meta,
```

Comment:
Nope, similar to `sidFor` it should call `metaFor` with the provided client_id and meta object. It should also assert that the client_id was provided in the clients array.